Remove `text` value for CSS colors

nt1m opened the pull request:
#### dc0028a55f2826bd4dbcda47cd2160e65f682e34
<pre>
Remove `text` value for CSS colors
<a href="https://bugs.webkit.org/show_bug.cgi?id=245626">https://bugs.webkit.org/show_bug.cgi?id=245626</a>
<a href="https://rdar.apple.com/100364547">rdar://100364547</a>

Reviewed by Abrar Rahman Protyasha.

`color: text` is non standard, and is equivalent to CanvasText. WebKit is the only browser engine that supports it.

Reject it at parse-time behind a linked on or after check for Cocoa platforms for backwards compat with apps, remove entire support for other platforms.

* LayoutTests/css-dark-mode/color-scheme-css.html:
* LayoutTests/css-dark-mode/color-scheme-meta.html:
* LayoutTests/css-dark-mode/color-scheme-priority.html:
* LayoutTests/css-dark-mode/older-syntax/supported-color-schemes-css.html:
* LayoutTests/css-dark-mode/older-syntax/supported-color-schemes-meta.html:
* LayoutTests/css-dark-mode/older-systems/color-scheme-css.html:
* LayoutTests/css-dark-mode/older-systems/color-scheme-meta.html:
* LayoutTests/css-dark-mode/older-systems/prefers-color-scheme.html:
* LayoutTests/css-dark-mode/prefers-color-scheme.html:
* LayoutTests/css3/color-filters/color-filter-ignore-semantic-expected.html:
* LayoutTests/css3/color-filters/color-filter-ignore-semantic.html:
* LayoutTests/fast/css/color-text-value-not-supported-expected.txt: Added.
* LayoutTests/fast/css/color-text-value-not-supported.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-shorthand-serialization-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-valid-expected.txt:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/css/values/color/CSSKeywordColor.cpp:
(WebCore::CSS::isSystemColorKeyword):
(WebCore::CSS::isDeprecatedSystemColorKeyword):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::systemColor const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::systemColor const):
* Websites/bugs.webkit.org/PrettyPatch/PrettyPatch.rb:

Canonical link: <a href="https://commits.webkit.org/317687@main">https://commits.webkit.org/317687@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/25822974f04a2b47bfdb3dcd8cc00c6298db3b74

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/176006 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/49939 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/43723 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/185600 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/138221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a45721a3-3fbc-4ebb-8a54-c638a7071ca7) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/177869 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/51231 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/49621 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/137065 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/138221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/320a6d2b-2419-4116-8140-99cfad6a19a6) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/178962 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/40534 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/157834 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/117544 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/084510e6-4ecd-4db9-8ee3-651c9b72d4cd) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/39177 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/37552 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 jsc-wpe](https://ews-build.webkit.org/#/builders/JSC-Tests-WPE-EWS "Waiting to run tests") | | 
| [✅ 🧪 jsc-x86-64](https://ews-build.webkit.org/#/builders/218/builds/6343 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/149595 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/37326 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/34069 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-debug-arm64](https://ews-build.webkit.org/#/builders/171/builds/37270 "Built successfully and passed tests") | [  ~~🛠 ios-safer-cpp~~](https://ews-build.webkit.org/#/builders/174/builds/41642 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/41758 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/188021 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/49606 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/157385 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/146076 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/39294 "Built successfully and passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/50287 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/170/builds/33951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/145908 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/40684 "Passed tests") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/122482 "Found 1 new failure in css/values/color/CSSKeywordColor.cpp") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/116360 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/48793 "Built successfully") | [✅ 🧪 mac-site-isolation](https://ews-build.webkit.org/#/builders/185/builds/12307 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/48195 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/48427 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/48252 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m committed:
Remove `text` value for CSS colors
https://bugs.webkit.org/show_bug.cgi?id=245626
rdar://100364547

Reviewed by NOBODY (OOPS!).

`color: text` is non standard, and is equivalent to CanvasText. WebKit is the only browser engine that supports it.

Reject it at parse-time behind a linked on or after check for Cocoa platforms for backwards compat with apps, remove entire support for other platforms.

* LayoutTests/css-dark-mode/color-scheme-css.html:
* LayoutTests/css-dark-mode/color-scheme-meta.html:
* LayoutTests/css-dark-mode/color-scheme-priority.html:
* LayoutTests/css-dark-mode/older-syntax/supported-color-schemes-css.html:
* LayoutTests/css-dark-mode/older-syntax/supported-color-schemes-meta.html:
* LayoutTests/css-dark-mode/older-systems/color-scheme-css.html:
* LayoutTests/css-dark-mode/older-systems/color-scheme-meta.html:
* LayoutTests/css-dark-mode/older-systems/prefers-color-scheme.html:
* LayoutTests/css-dark-mode/prefers-color-scheme.html:
* LayoutTests/css3/color-filters/color-filter-ignore-semantic-expected.html:
* LayoutTests/css3/color-filters/color-filter-ignore-semantic.html:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/css/values/color/CSSKeywordColor.cpp:
(WebCore::CSS::isSystemColorKeyword):
(WebCore::CSS::isDeprecatedSystemColorKeyword):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::systemColor const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::systemColor const):
* Websites/bugs.webkit.org/PrettyPatch/PrettyPatch.rb:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c2c9137ec0d043b4cf39ccfec9d5f4d2f70ea2a9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50118

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5d9f697ba53e4f7d60b9fe7285bcb1fee63b4b04. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50118

nt1m committed:
Remove `text` value for CSS colors
https://bugs.webkit.org/show_bug.cgi?id=245626
rdar://100364547

Reviewed by Abrar Rahman Protyasha.

`color: text` is non standard, and is equivalent to CanvasText. WebKit is the only browser engine that supports it.

Reject it at parse-time behind a linked on or after check for Cocoa platforms for backwards compat with apps, remove entire support for other platforms.

* LayoutTests/css-dark-mode/color-scheme-css.html:
* LayoutTests/css-dark-mode/color-scheme-meta.html:
* LayoutTests/css-dark-mode/color-scheme-priority.html:
* LayoutTests/css-dark-mode/older-syntax/supported-color-schemes-css.html:
* LayoutTests/css-dark-mode/older-syntax/supported-color-schemes-meta.html:
* LayoutTests/css-dark-mode/older-systems/color-scheme-css.html:
* LayoutTests/css-dark-mode/older-systems/color-scheme-meta.html:
* LayoutTests/css-dark-mode/older-systems/prefers-color-scheme.html:
* LayoutTests/css-dark-mode/prefers-color-scheme.html:
* LayoutTests/css3/color-filters/color-filter-ignore-semantic-expected.html:
* LayoutTests/css3/color-filters/color-filter-ignore-semantic.html:
* LayoutTests/fast/css/color-text-value-not-supported-expected.txt: Added.
* LayoutTests/fast/css/color-text-value-not-supported.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-shorthand-serialization-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-backgrounds/parsing/background-valid-expected.txt:
* Source/WTF/wtf/cocoa/RuntimeApplicationChecksCocoa.h:
* Source/WebCore/css/values/color/CSSKeywordColor.cpp:
(WebCore::CSS::isSystemColorKeyword):
(WebCore::CSS::isDeprecatedSystemColorKeyword):
* Source/WebCore/rendering/RenderTheme.cpp:
(WebCore::RenderTheme::systemColor const):
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::systemColor const):
* Websites/bugs.webkit.org/PrettyPatch/PrettyPatch.rb:

Canonical link: https://commits.webkit.org/317687@main
